core/src/main/java/com/orientechnologies/orient/core/db/document/ODatabaseDocumentAbstract.java

```diff
@@ -1429,17 +1429,17 @@ public void release() {
    * Creates a new ODocument.
    */
   public ODocument newInstance() {
-    return new ODocument();
+    return new ODocument(this);
```

Comment:
What is the reason to do this? 
This thing scares me a lot... there is always a chance that the user does a `new ODocument().save()`

Reply:
this is just removing an access to the thread local, inside the `new ODocument()` we access to the tread local to get the current database, i just moved the lookup of the database from the thread local to the API itself.

Reply:
OK, but what if you close the DB and then save the document with another DB? 
We are changing the behaviour and breaking backward compatibility

Reply:
Oh ok, I got it, never mind, it's good ;-)